common/cards/card-plugins/single-use/target-block.js

```diff
@@ -30,15 +31,24 @@ class TargetBlockSingleUseCard extends SingleUseCard {
 	 */
 	onApply(game, instance, pos, pickedSlots) {
 		const {player} = pos
+		const setWeakness = this.getInstanceKey(instance, 'setWeakness')
 		const pickedSlot = pickedSlots[this.id]?.[0]
 		if (!pickedSlot) return
 
 		player.hooks.beforeAttack[instance] = (attack) => {
-			console.log(attack.type)
-			if (['backlash', 'ailment'].includes(attack.type)) return
+			if (attack.type === 'weakness' && !player.custom[setWeakness]) {
+				attack.multiplyDamage(0).lockDamage
+			}
+			delete player.custom[setWeakness]
+
+			if (['backlash', 'ailment', 'weakness'].includes(attack.type)) return
 			if (!pickedSlot.row || !pickedSlot.row.state.hermitCard) return
 			attack.target.rowIndex = pickedSlot.row.index
 			attack.target.row = pickedSlot.row.state
+
+			const weaknessAttack = createWeaknessAttack(attack)
```

Comment:
You are creating weakness attacks for effect attacks, this probably should be looking for a weakness attack, if it found one create a new one and if createWeaknessAttack generated an attack then reduce the old one to 0 and add the new attack.